discovery_engine_core/core/src/engine.rs

```diff
@@ -342,20 +352,24 @@ impl Engine<xayn_ai::ranker::Ranker> {
                 .map_err(|err| Error::Ranker(err.into()))?
                 .build()
                 .map_err(|err| Error::Ranker(err.into()))?;
-            Engine::from_state(&state.engine, config.into(), ranker, stacks_ops).await
+            Engine::from_state(&state.engine, config.into(), ranker, stack_ops).await
         } else {
             let ranker = builder.build().map_err(|err| Error::Ranker(err.into()))?;
-            Engine::new(config.into(), ranker, stacks_ops).await
+            Engine::new(config.into(), ranker, stack_ops).await
         }
     }
 }
 
+/// Locked discovery engine.
+#[allow(clippy::module_name_repetitions)]
+pub struct LockedEngine(pub Mutex<Engine<xayn_ai::ranker::Ranker>>);
```

Comment:
`Locked` is often seen as "has a the lock" so maybe `SharedEngine` or similar would be better.